### PR TITLE
Callout component

### LINKED ISSUE #359: [FEATURE] Callout component
Add a new Callour component, following design and CSS styling as shown in [Bootstrap Italia documentation page](https://italia.github.io/bootstrap-italia/docs/componenti/callout/).

### PR DESCRIPTION
**Pull request by daniel-234**
Fixes #359 

#### PR Checklist

- [x] My branch is up-to-date with the Upstream `master` branch.
- [ ] The unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
- Add the Callout component
- Add stories for the Callout component

### CODE REVIEW

File: src/components/Callout/Callout.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string])
+}
+
+const defaultProps = {
+  color: '',
+  highlight: false,
+  tag: 'div'
+}
+
+const Callout = ({ children, color, highlight, tag: Tag }) => {
```
Comment (dej611) [inline src/components/Callout/Callout.js]: Add `...attributes` here

File: src/components/Callout/Callout.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  const classes = classNames('callout', color, {
+    'callout-highlight': highlight
+  })
+  return <Tag className={classes}>{children}</Tag>
```
Comment (dej611) [inline src/components/Callout/Callout.js]: add the `...attributes` here

File: src/components/Callout/CalloutText.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  children: PropTypes.node,
+  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string])
+}
+
+const defaultProps = {
+  bigText: false,
+  tag: 'p'
+}
+
+const CalloutText = ({ bigText, children, className, tag: Tag }) => {
```
Comment (dej611) [inline src/components/Callout/CalloutText.js]: add the `...attributes` here

File: src/components/Callout/CalloutText.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  const classes = classNames(className, {
+    'callout-big-text': bigText
+  })
+  return <Tag className={classes}>{children}</Tag>
```
Comment (dej611) [inline src/components/Callout/CalloutText.js]: add `...attributes` here

File: src/components/Callout/CalloutTitle.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  children: PropTypes.node,
+  className: PropTypes.string,
+  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string])
+}
+
+const defaultProps = {
+  tag: 'div'
+}
+
+const CalloutTitle = ({ children, className, tag: Tag }) => {
```
Comment (dej611) [inline src/components/Callout/CalloutTitle.js]: add `...attributes` here

File: src/components/Callout/CalloutTitle.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  const classes = classNames(className, 'callout-title')
+  return <Tag className={classes}>{children}</Tag>
```
Comment (dej611) [inline src/components/Callout/CalloutTitle.js]: add `...attributes` here

File: stories/Callout/Callout.stories.js [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+      varius.
+    </CalloutText>
+  </Callout>
+)
+
+const ExampleWithIcon = () => (
+  <Callout>
+    <CalloutTitle>
+      <Icon
+        className={'aria-hidden: true'}
```
Comment (dej611) [inline stories/Callout/Callout.stories.js]: this doesn't seem ok. Do you mean `style={{ariaHidden: true}}` maybe?

File: stories/Callout/Callout.stories.js [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+        color=""
+        icon="it-info-circle"
+        padding={false}
+        size=""
+      />
```
Comment (dej611) [inline stories/Callout/Callout.stories.js]: If props are not used you can just drop them: 
```jsx
<Icon icon="info-circle" padding={false} style={{ariaHidden: true}} />
```

File: stories/Callout/Callout.stories.js [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+const ExampleForScreenReader = () => (
+  <Callout>
+    <CalloutTitle>
+      <Icon
+        className={'aria-hidden: true'}
+        color=""
+        icon="it-check-circle"
+        padding={false}
+        size=""
+      />
```
Comment (dej611) [inline stories/Callout/Callout.stories.js]: See above

File: stories/Callout/Callout.stories.js [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+const CalloutSuccess = () => (
+  <Callout color={'success'}>
+    <CalloutTitle>
+      <Icon
+        className={'aria-hidden: true'}
+        color=""
+        icon="it-check-circle"
+        padding={false}
+        size=""
+      />
```
Comment (dej611) [inline stories/Callout/Callout.stories.js]: See above

File: stories/Callout/Callout.stories.js [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+const CalloutWarning = () => (
+  <Callout color={'warning'}>
+    <CalloutTitle>
+      <Icon
+        className={'aria-hidden: true'}
+        color=""
+        icon="it-help-circle"
+        padding={false}
+        size=""
+      />
```
Comment (dej611) [inline stories/Callout/Callout.stories.js]: See above

File: stories/Callout/Callout.stories.js [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+const CalloutDanger = () => (
+  <Callout color={'danger'}>
+    <CalloutTitle>
+      <Icon
+        className={'aria-hidden: true'}
+        color=""
+        icon="it-close-circle"
+        padding={false}
+        size=""
+      />
```
Comment (dej611) [inline stories/Callout/Callout.stories.js]: See above

File: stories/Callout/Callout.stories.js [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+const CalloutImportant = () => (
+  <Callout color={'important'}>
+    <CalloutTitle>
+      <Icon
+        className={'aria-hidden: true'}
+        color=""
+        icon="it-info-circle"
+        padding={false}
+        size=""
+      />
```
Comment (dej611) [inline stories/Callout/Callout.stories.js]: See above

File: stories/Callout/Callout.stories.js [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+const CalloutNote = () => (
+  <Callout color={'note'}>
+    <CalloutTitle>
+      <Icon
+        className={'aria-hidden: true'}
+        color=""
+        icon="it-info-circle"
+        padding={false}
+        size=""
+      />
```
Comment (dej611) [inline stories/Callout/Callout.stories.js]: See above

File: stories/Callout/Callout.stories.js (hunk 7 above)
Comment (daniel-234) [reply stories/Callout/Callout.stories.js]: Yes, I did. Thanks for pointing it out.

**Comment by daniel-234** [comment]
Thanks for your review. I'm adding a commit with the changes from your suggestions.

**Comment by dej611** [comment]
Ok, code is good.
Last touch is to change the icon prefix name: `it-<icon-name>` => `<icon-name>`.
For instance:

```jsx
<Icon
 icon="it-info-circle"
 padding={false}
 style={{ ariaHidden: true }}
/>
```

becomes:

```jsx
<Icon
 icon="info-circle"
 padding={false}
 style={{ ariaHidden: true }}
/>
```

The `it` prefix is already managed internally in the component.

**Pull request merged**